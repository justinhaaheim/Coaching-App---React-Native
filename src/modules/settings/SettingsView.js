/**
  Name: SettingsView

  Description:

  TODO:

  Copyright (c) 2017-present Justin Haaheim

  This file is subject to the terms and conditions defined in
  file 'LICENSE', which is part of this source code package.

********************************************** */

import React, { PropTypes, Component } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  //  Text,
  View,
  //  Switch,
  //  List,
  //  ListItem,
} from 'react-native';

import NotificationsIOS from 'react-native-notifications';

import {
  Container,
  Content,
  List,
  ListItem,
  Text,
  Icon,
  Badge,
  Left,
  Body,
  Right,
  Separator,
  StyleProvider,
  Switch,
} from 'native-base';

import getTheme from '../../native-base-theme/components';
// import material from '../../native-base-theme/variables/material';

import MaterialIcon from 'react-native-vector-icons/MaterialIcons';


class SettingsView extends Component {
  // What does this do? Part of react-navigation ??
  static displayName = 'SettingsView';

  static navigationOptions = {
    title: 'Settings',
  };

  static propTypes = {
    mleEnabled: PropTypes.bool.isRequired,
    // liList: PropTypes.string.isRequired,
    settingsStateActions: PropTypes.shape({
      toggleArenaVersion: PropTypes.func.isRequired,
    }).isRequired,
    navigate: PropTypes.func.isRequired,
  };

  fireLocalNotification() {
    let localNotification = NotificationsIOS.localNotification({
      fireDate: new Date(Date.now() + (2 * 1000)).toISOString(),
        alertBody: "Local notificiation!",
        alertTitle: "Local Notification Title",
        alertAction: "Click here to open",
        soundName: "chime.aiff",
        foreground: true,
        category: "SOME_CATEGORY",
        userInfo: { }
      });
  }


  render() {
    const { toggleArenaVersion, toggleNotifications, updateLi } = this.props.settingsStateActions;
    const { mleEnabled, notificationsEnabled } = this.props;

    return (
      <StyleProvider style={getTheme()}>
        <Container style={styles.container}>
          <Content>
            <List style={styles.whiteBg}>

              <Separator bordered>
                <Text>
                  COACHING ARENA
                </Text>
              </Separator>

              <ListItem icon>
                <Left>
                  <Icon name="ios-list-box-outline" />
                </Left>
                <Body>
                  <Text>MLE/TCI Version</Text>
                </Body>
                <Right>
                  <Switch
                    onValueChange={ () =>
                      toggleArenaVersion()
                    }
                    value={mleEnabled} />
                </Right>
              </ListItem>

              <ListItem icon>
                <Left>
                  <Icon name="ios-notifications" />
                </Left>
                <Body>
                  <Text>Remind me to set the arena</Text>
                </Body>
                <Right>
                  <Switch
                    onValueChange={ () =>
                      toggleNotifications()
                    }
                    value={notificationsEnabled} />
                </Right>
              </ListItem>

              <Separator bordered />

              <ListItem
                icon
                onPress={ () => {
                  this.props.navigation.navigate('LiEntry', {
                    title: "Life's Intentions",
                    // content: () => liList, // Buggy - doesn't update after navigation?
                    onUpdate: updateLi,  // already bound to dispatch
                  });
                }}
                >
                <Left>
                  <Icon name="ios-body" />
                </Left>
                <Body>
                  <Text>Edit Life's Intentions</Text>
                </Body>
                <Right>
                  <Icon name="arrow-forward" />
                </Right>
              </ListItem>

              <Separator bordered />

              <ListItem
                icon
                onPress={ this.fireLocalNotification.bind(this) }
                >
                <Left>
                  <Icon name="ios-body" />
                </Left>
                <Body>
                  <Text>Test Notification</Text>
                </Body>
                <Right>
                  <Icon name="arrow-forward" />
                </Right>
              </ListItem>

              <Separator bordered />

              <ListItem
                icon
                onPress={ this.fireLocalNotification.bind(this) }
                >
                <Left>
                  <Icon name="ios-ribbon" />
                </Left>
                <Body>
                  <Text>Credits & Acknowledgements</Text>
                </Body>
                <Right>
                  <Icon name="arrow-forward" />
                </Right>
              </ListItem>
            </List>
          </Content>
        </Container>
    </StyleProvider>
    );
  }
}

const circle = {
  borderWidth: 0,
  borderRadius: 40,
  width: 80,
  height: 80,
};

const styles = {
  container: {
    // flex: 1,
    // flexDirection: 'column',
    // justifyContent: 'flex-start',
    // alignItems: 'stretch',
    // alignSelf: 'flex-start',
    backgroundColor: '#F0EFF5'
  },
  whiteBg: {
    backgroundColor: 'white',
  },
};

export default SettingsView;
