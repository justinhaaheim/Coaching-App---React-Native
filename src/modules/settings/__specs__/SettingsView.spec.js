import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import SettingsView from '../SettingsView';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <SettingsView
        mleEnabled={false}
        notificationsEnabled={false}
        settingsStateActions={{
          toggleArenaVersion: () => 'Toggle Arena Version',
          toggleNotifications: () => 'Toggle Notifications',
        }}
        navigation={{
          navigate: () => 'Navigate'
        }}
        buttons={[{ name: 'Test Button', selected: false }]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
