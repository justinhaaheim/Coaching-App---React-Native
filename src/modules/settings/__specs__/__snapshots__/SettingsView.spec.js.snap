// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#efeff3",
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        virtual={undefined}
      >
        <View
          bordered={true}
          style={
            Object {
              "backgroundColor": "#efeff3",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "borderTopWidth": 0.5,
              "flex": 1,
              "height": 35,
              "justifyContent": "center",
              "paddingBottom": 11,
              "paddingLeft": 15,
              "paddingTop": 11,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 14,
              }
            }
            virtual={undefined}
          >
            COACHING ARENA
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            icon={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "borderBottomWidth": null,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "marginLeft": 15,
                "paddingBottom": null,
                "paddingRight": null,
                "paddingTop": null,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "center",
                  "paddingRight": 15,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 28,
                      "width": 25,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 17,
                    "marginLeft": null,
                    "marginRight": 10,
                  }
                }
                virtual={undefined}
              >
                MLE/TCI Version
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "flex-end",
                  "paddingBottom": null,
                  "paddingLeft": null,
                  "paddingRight": 15,
                  "paddingTop": null,
                }
              }
              virtual={undefined}
            >
              <RCTSwitch
                disabled={false}
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onValueChange={[Function]}
                style={
                  Array [
                    Object {
                      "height": 31,
                      "width": 51,
                    },
                    Object {
                      "alignSelf": null,
                      "marginBottom": -5,
                      "marginRight": undefined,
                      "marginTop": -5,
                    },
                  ]
                }
                value={false}
                virtual={undefined}
              />
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            icon={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "borderBottomWidth": null,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "marginLeft": 15,
                "paddingBottom": null,
                "paddingRight": null,
                "paddingTop": null,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "center",
                  "paddingRight": 15,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 28,
                      "width": 25,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 17,
                    "marginLeft": null,
                    "marginRight": 10,
                  }
                }
                virtual={undefined}
              >
                Remind me to set the arena
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderBottomWidth": 0,
                  "borderColor": "#c9c9c9",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "flex-end",
                  "paddingBottom": null,
                  "paddingLeft": null,
                  "paddingRight": 15,
                  "paddingTop": null,
                }
              }
              virtual={undefined}
            >
              <RCTSwitch
                disabled={false}
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onValueChange={[Function]}
                style={
                  Array [
                    Object {
                      "height": 31,
                      "width": 51,
                    },
                    Object {
                      "alignSelf": null,
                      "marginBottom": -5,
                      "marginRight": undefined,
                      "marginTop": -5,
                    },
                  ]
                }
                value={false}
                virtual={undefined}
              />
            </View>
          </View>
        </View>
        <View
          bordered={true}
          style={
            Object {
              "backgroundColor": "#efeff3",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "borderTopWidth": 0.5,
              "flex": 1,
              "height": 35,
              "justifyContent": "center",
              "paddingBottom": 11,
              "paddingLeft": 15,
              "paddingTop": 11,
            }
          }
          virtual={undefined}
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            icon={true}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "borderBottomWidth": null,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "marginLeft": 15,
                "paddingBottom": null,
                "paddingRight": null,
                "paddingTop": null,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "center",
                  "paddingRight": 15,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 28,
                      "width": 25,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 17,
                    "marginLeft": null,
                    "marginRight": 10,
                  }
                }
                virtual={undefined}
              >
                Edit Life's Intentions
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderBottomWidth": 0,
                  "borderColor": "#c9c9c9",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "flex-end",
                  "paddingBottom": null,
                  "paddingLeft": null,
                  "paddingRight": 15,
                  "paddingTop": null,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#C8C7CC",
                      "fontSize": 20,
                      "paddingLeft": 10,
                      "paddingTop": 3,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          bordered={true}
          style={
            Object {
              "backgroundColor": "#efeff3",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "borderTopWidth": 0.5,
              "flex": 1,
              "height": 35,
              "justifyContent": "center",
              "paddingBottom": 11,
              "paddingLeft": 15,
              "paddingTop": 11,
            }
          }
          virtual={undefined}
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            icon={true}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "borderBottomWidth": null,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "height": 44,
                "justifyContent": "center",
                "marginLeft": 15,
                "paddingBottom": null,
                "paddingRight": null,
                "paddingTop": null,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "center",
                  "paddingRight": 15,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 28,
                      "width": 25,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 17,
                    "marginLeft": null,
                    "marginRight": 10,
                  }
                }
                virtual={undefined}
              >
                Credits & Acknowledgements
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderBottomWidth": 0,
                  "borderColor": "#c9c9c9",
                  "flex": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "flex-end",
                  "paddingBottom": null,
                  "paddingLeft": null,
                  "paddingRight": 15,
                  "paddingTop": null,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "#C8C7CC",
                      "fontSize": 20,
                      "paddingLeft": 10,
                      "paddingTop": 3,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          bordered={true}
          style={
            Object {
              "backgroundColor": "#efeff3",
              "borderBottomWidth": 0,
              "borderColor": "#c9c9c9",
              "borderTopWidth": 0.5,
              "flex": 1,
              "height": 35,
              "justifyContent": "center",
              "paddingBottom": 11,
              "paddingLeft": 15,
              "paddingTop": 11,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={null}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 20,
        }
      }
    >
      <View
        danger={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.8)",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        success={false}
        virtual={undefined}
        warning={false}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "flex": 1,
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#757575",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
        <RCTScrollView
          dataArray={Array []}
          dataSource={
            ListViewDataSource {
              "items": 0,
            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
            Object {
              "marginHorizontal": -15,
              "marginTop": 15,
            }
          }
          virtual={undefined}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
